occurrences/schema.py

```diff
@@ -406,8 +406,9 @@ def validate_enrolment(study_group, occurrence, new_enrolment=True):
     if occurrence.cancelled:
         raise EnrolCancelledOccurrenceError("Cannot enrol cancelled occurrence")
     if (
-        study_group.group_size > occurrence.max_group_size
-        or study_group.group_size < occurrence.min_group_size
+        occurrence.max_group_size and study_group.group_size > occurrence.max_group_size
```

Comment:
After this change, if `max_group_size` is `0` the check will be skipped, which is probably not what you want.

Reply:
Good point, but that'd be something we want. This validation should be run only if the field value > 0. Still made a change to improve readability